src/components/Wizards/CreateVmWizard/tests/NetworksTab.test.js

```diff
@@ -1,13 +1,116 @@
   });
+  it('passes existing networks to FormFactory', () => {
+    const props = { ...NetworksTabFixture.props };
+    props.networks = defaultNetworks;
+    const component = testNetworksTab(props);
+
+    const passedRows = component.find(TableFactory).props().rows;
+    expect(passedRows).toHaveLength(3);
+    expect(passedRows[0].addendum).toBeNull();
+    expect(passedRows[0].renderConfig).toEqual(1);
+    expect(passedRows[1].addendum).toEqual('(Bootable)');
+    expect(passedRows[1].isBootable).toBeTruthy();
+    expect(passedRows[1].renderConfig).toEqual(0);
+    expect(passedRows[2].addendum).toBeNull();
+    expect(passedRows[2].isBootable).toBeFalsy();
+    expect(passedRows[2].renderConfig).toEqual(0);
+  });
+  it('checkPxeBootNetwork', () => {
+    const props = { ...NetworksTabFixture.props };
+    props.networks = defaultNetworks;
+    const component = testNetworksTab(props);
+
+    const { rows } = component.state();
+    expect(component.instance().checkPxeBootNetwork(rows)).toBeFalsy();
+
+    expect(component.instance().checkPxeBootNetwork(rows.slice(0, 1))).toBeTruthy();
+
+    props.pxeBoot = false;
+    component.setProps(props);
+    expect(component.instance().checkPxeBootNetwork(rows)).toBeFalsy();
+    expect(component.instance().checkPxeBootNetwork(rows.slice(0, 1))).toBeFalsy();
+  });
+  it('resolveBootability', () => {
+    const props = { ...NetworksTabFixture.props };
+    props.networks = defaultNetworks;
+    const component = testNetworksTab(props);
+
+    const { rows } = component.state();
+    // change rows order
+    let newRows = component.instance().resolveBootability([rows[0], rows[2], rows[1]]);
+    expect(newRows[0].isBootable).toBeFalsy();
+    expect(newRows[0].addendum).toBeNull();
+    expect(newRows[1].isBootable).toBeTruthy();
+    expect(newRows[1].addendum).toEqual('(Bootable)');
+    expect(newRows[2].isBootable).toBeFalsy();
+    expect(newRows[2].addendum).toBeNull();
+
+    newRows = component.instance().resolveBootability([rows[1], rows[0], rows[2]]);
+    expect(newRows[0].isBootable).toBeTruthy();
+    expect(newRows[0].addendum).toEqual('(Bootable)');
+    expect(newRows[1].isBootable).toBeFalsy();
+    expect(newRows[1].addendum).toBeNull();
+    expect(newRows[2].isBootable).toBeFalsy();
+    expect(newRows[2].addendum).toBeNull();
+  });
+  it('create new network', () => {
+    const props = { ...NetworksTabFixture.props };
+    props.networks = defaultNetworks.slice(0, 1);
+    const component = testNetworksTab(props, true);
+    const { nextId, rows, editing } = component.state();
+
+    expect(editing).toBeFalsy();
+
+    component
+      .find('#create-network-btn')
+      .at(1)
+      .simulate('click');
+
+    expect(component.state().editing).toBeTruthy();
+    expect(component.state().nextId).toEqual(nextId + 1);
+    expect(component.state().rows).toHaveLength(rows.length + 1);
+
+    const newRow = component.state().rows[rows.length];
```

Comment:
I don't think it is necessary to look at the inner state of the component when we can just check the `onChange` callback